services/ble/Gap.hpp

```diff
@@ -285,6 +272,47 @@ namespace services
         return static_cast<GapPeripheral::AdvertisementFlags>(infra::enum_cast(lhs) | infra::enum_cast(rhs));
     }
 
+    class GapAdvertisingDataParser
+    {
+    public:
+        explicit GapAdvertisingDataParser(infra::ConstByteRange data);
+
+        infra::ConstByteRange LocalName() const;
+        infra::Optional<std::pair<uint16_t, infra::ConstByteRange>> ManufacturerSpecificData() const;
+        infra::Optional<GapPeripheral::AdvertisementFlags> Flags() const;
+        infra::MemoryRange<AttAttribute::Uuid16> CompleteListOf16BitUuids() const;
+        infra::MemoryRange<AttAttribute::Uuid128> CompleteListOf128BitUuids() const;
+
+    private:
+        infra::ConstByteRange data;
+
+    private:
+        infra::ConstByteRange ParserAdvertisingData(GapAdvertisementDataType type) const;
+    };
+
+    class GapAdvertisementFormatter
+    {
+    public:
+        explicit GapAdvertisementFormatter(infra::BoundedVector<uint8_t>& payload);
+
+        void AppendFlags(GapPeripheral::AdvertisementFlags flags);
+        void AppendCompleteLocalName(const infra::BoundedConstString& name);
+        void AppendShortenedLocalName(const infra::BoundedConstString& name);
+        void AppendManufacturerData(uint16_t manufacturerCode, infra::ConstByteRange data);
+        void AppendListOfServicesUuid(infra::MemoryRange<AttAttribute::Uuid16> services);
+        void AppendListOfServicesUuid(infra::MemoryRange<AttAttribute::Uuid128> services);
+        void AppendPublicTargetAddress(hal::MacAddress address);
+        void Clear();
```

Comment:
Why is Clear() needed? In practical cases, you should be able to reconstruct the storage (or invoke clear() on the BoundedVector). A formatter should be single-use.